### ISSUE TITLE
Segmentation

### ISSUE THREAD

**Issue by balthazarneveu** [balthazarneveu/geosciences#1]
https://wandb.ai/balthazarneveu/geosciences-segmentation

- [x] LR scheduler (plateau)
- [x] Validation accuracy
- [x] Validation Metric IoU, dice coeff
- [x] Augmentations (vertical reverse, horizontal wrap)
- [x] Wrapped convolutions
- [x] Add pos_weight in BCE to give stronger importance to segmented areas 
- [x] Flexible Stacked Conv
- [x] Flexible Stacked Conv + Residual connections
- [x] Review input dynamics (remove NaN)
- [x] New metrics logging (Recall Precision F1 score)
- [x] Add pretrained models
- [x] New dataset + update kaggle version
- [x] Could not optimize  Dice loss (1-dice coeff) 
- [x] Fix DICE loss computation
- [x] Show dice/metrics in interactive pipe
- [x] Control zero labels images.
- [x] UNET+circular repeat = pad in + crop out
- [ ] Re-Split
- [x] Fix metrics normalization 
- [ ] Augmentations (add noise, shift dynamic range, blur, sharpen, shear?)
- [ ] Review input dynamics (seems like clipping values to (-0.05 ,0.05) would be a good idea)

-------
Lightweight models
- [x] Knowledge distillation
- [x] Report on KD

-------
Model performances
- [ ] Refine annotations "super pixels"
- [ ] Refine annotations SAM

------
Bonus 
- [ ] Mine difficult examples
- [ ] Attention layer

**Comment by balthazarneveu** [comment]
![image](https://github.com/balthazarneveu/geosciences/assets/41070742/69ff1a10-df24-4f8f-be55-8866e8cf7b22)
Deal with wrapping

All ones out: Loss: 1.217e+00 Accuracy: 9.639%
All minus one: Loss 4.096e-01 Accuracy: 90.361%

**Comment by balthazarneveu** [comment]
- [x] Static quantization
- [ ] Knowledge distillation on Kaggle requires loading models out there!
- [x] Knowledge distillation framework
- [x] temperature scaling
- [ ] QAT

![image](https://github.com/balthazarneveu/geosciences/assets/41070742/9b97ed15-e9f4-42a4-ab29-718d33325cd4)

**Comment by balthazarneveu** [comment]
![image](https://github.com/balthazarneveu/geosciences/assets/41070742/db6ff8d3-b73c-4325-8dcd-d78d2735a2b6)
